src/components/TableNew/TableHeaderCell.stories.tsx

```diff
@@ -0,0 +1,144 @@
+import type { Meta, StoryObj } from '@storybook/react'
+import { AxeIcon, PiIcon } from 'lucide-react'
+import { useState } from 'react'
+import { TableHeaderCell, TableHeaderNode } from '.'
+import { Header } from '../TypographyNew'
+
+const meta: Meta<typeof TableHeaderCell> = {
+  title: 'Components/TableNew/TableHeaderCell',
+  component: TableHeaderCell,
+  tags: ['autodocs'],
+}
+
+export default meta
+
+type Story = StoryObj<typeof TableHeaderCell>
+
+export const Default: Story = {
+  render: () => {
+    const [clicks, setClicks] = useState(0)
+
+    const handleClick = () => {
+      setClicks(prev => prev + 1)
+      console.log('clicked')
+    }
+
+    return (
+      <div className="flex flex-col gap-4 w-[300px] bg-red-200 text-green-800">
+        <TableHeaderCell
+          id="some-id"
+          onClick={handleClick}
+          data-testid="table-header-cell"
+          className="outline outline-amber-400 cursor-pointer hover:bg-amber-100 transition-color"
+        >
+          <TableHeaderNode>
+            <Header variant="h1">Some title</Header>
+            <Header variant="h2">And a subtitle</Header>
+          </TableHeaderNode>
+        </TableHeaderCell>
+        {clicks > 0 && (
+          <div className="text-sm text-black font-medium">✓ Header cell was clicked {clicks} times!</div>
+        )}
+      </div>
+    )
+  },
+}
+
+export const WithIcon: Story = {
+  render: () => {
+    const [isClicked, setIsClicked] = useState(false)
+
+    const handleClick = () => {
+      setIsClicked(prev => !prev)
+    }
+
+    return (
+      <>
+        <TableHeaderCell id="some-id" onClick={handleClick} className="outline outline-amber-400 w-[300px]">
+          {isClicked ? <PiIcon /> : <AxeIcon />}
+          <TableHeaderNode>
+            <Header variant="h1">Some title</Header>
+            <Header variant="h2">And a subtitle</Header>
+          </TableHeaderNode>
+        </TableHeaderCell>
+      </>
+    )
+  },
+}
+
+export const CanSwapContent: Story = {
```

Comment:
I'm not sure about the value of this story.

Reply:
It was more meant as a guide to usage. That is to show that the content can be replaced without changing the cell itself. I can delete it, but honestly, this is just a place where we can see all sorts of usage of the item. I don't think that we could have too many (reasonably). What is going to happen?